advanced_stats_starter.py

```diff
@@ -47,6 +48,101 @@ def get_adv_season_stats():
     adv_stat_team_season = adv_stat_team_season.drop(['LTeamID'], axis=1)
     adv_stat_team_season = adv_stat_team_season.rename(columns={'WTeamID': 'TeamID'})
 
+    print(adv_stat_team_season.head())
+
     return adv_stat_team_season
 
-print(get_adv_season_stats().head())
+
+def get_adv_stats_per_season_game():
+    seasons_results = pd.read_csv("input/RegularSeasonDetailedResults.csv")
+    seasons_results = seasons_results.sort_values(['Season', 'DayNum'])
+
+    team_stats = {}
+    rating_per_game = pd.DataFrame()
+    print(seasons_results.shape)
+    print(rating_per_game.dtypes)
+    for index, row in seasons_results.iterrows():
+        # find rating for WTeamID first
+        print(index)
+        isCurrentSeason = seasons_results['Season'] == row.Season
+        isPrevGames = seasons_results['DayNum'] < row.DayNum
```

Comment:
I think it should include the current game as well. The advanced stats rating should be up to that game

Reply:
but you are using previous season game to calculate the current rating for each team and **predict the outcome of the current game**

Also including current game doesn't work because we are going to get playoff game result to calculate their rating when predict if they are going to win or not

Reply:
My understanding is we are using regular season stats to train. Partly because there are more games to train on vses using playoff data to train. Once the model is trained, we predict the playoff games by using their end of regular season advanced stats. When we train, we think of the teams advanced stats up to that point of the season and how the advanced stats correlate to winning or losing. For prediction, we just calculate the teams end of regular season stats and predict.

Reply:
yea that correct, isn't that what we are doing here 😆